index.bs

```diff
@@ -432,10 +434,29 @@ the computation graph used to compute one or more specified outputs. The key
 purpose of the compilation step is to enable optimizations that span two or
 more operations, such as operation or loop fusion.
 
-The {{MLGraph/compute()}} method of the {{MLGraph}} interface is used to execute the
-compiled computation graph (to perform inference). The caller supplies the input
-values using {{MLNamedInputs}}, binding the input {{MLOperand}}s to their values.
-The caller supplies pre-allocated buffers for output {{MLOperand}}s using {{MLNamedOutputs}}.
+Once the {{MLGraph}} is constructed, there are multiple ways by which the graph may be executed.
+The {{MLExecution}} interface represents a way the execution of the graph is carried out immediately on the
+calling thread, which must also be a worker thread. The execution is carried out by the {{MLExecution/compute()}},
+method which produces the results of the computation from all the inputs bound to graph on the bound outputs.
+This type of execution is limited to only when the computational device bound to the ML context is a CPU.
+
+The {{MLAwaitedExecution}} interface represents a way the execution of the graph is performed asynchronously
+on a separate worker thread. The call to the {{MLAwaitedExecution/compute()}} method returns instantly without 
```

Comment:
```suggestion
in the background. The call to the {{MLAwaitedExecution/compute()}} method returns instantly without 
```
or maybe just say "separate thread" (but "worker thread" makes it feel that it will be run in a developer-exposed worker)